lib/active_merchant/billing/gateways/blue_snap.rb

```diff
@@ -237,6 +238,22 @@ def add_address(doc, options)
         doc.zip(address[:zip]) if address[:zip]
       end
 
+      def add_3ds(doc, three_d_secure_options)
+        eci = three_d_secure_options[:eci]
+        cavv = three_d_secure_options[:cavv]
+        xid = three_d_secure_options[:xid]
+        ds_transaction_id = three_d_secure_options[:ds_transaction_id]
+        version = three_d_secure_options[:version]
+
+        doc.send('three-d-secure') do
```

Comment:
👍  the conditional when `add_3ds` is called prevents sending an empty `three-d-secure` element.